### PR TITLE
Added single output mode to dispatch component

### PR DESCRIPTION
**Pull request by Arivias**
[FEATURE][WIKI]

## About the PR 
Adds a new option to the dispatch component: Toggle single output mode.
When active, the dispatch component will only pass a signal through the first route that successfully validates.

## Why's this needed? 
Mechcomp doesn't have an easy way to do if-elseif-else.

## Changelog

```changelog
(u)Sidewinder7
(*)Added a single-output mode to the mechcomp dispatch component.
```

### CODE REVIEW

File: code/WorkInProgress/MechanicMadness.dm [hunk 1]
```diff
@@ -1465,14 +1465,15 @@
 	desc = ""
 	icon_state = "comp_disp"
 	var/exact_match = 0
+	var/single_output = 0
```
Comment (ZeWaka) [inline code/WorkInProgress/MechanicMadness.dm:1468]:
```suggestion
	var/exact_match = FALSE
	var/single_output = FALSE
```

File: code/WorkInProgress/MechanicMadness.dm [hunk 2]
```diff
@@ -1529,13 +1537,13 @@
 	//Called when mechanics_holder tries to fire out signals
 	proc/runFilter(var/comsig_target, atom/receiver, var/signal)
 		if(!(receiver in src.outgoing_filters))
-			return 0 //Not filtering this output, let anything pass
+			return src.single_output? 2 : 0 //Not filtering this output, let anything pass
```
Comment (ZeWaka) [inline code/WorkInProgress/MechanicMadness.dm:1540]: Can we get some kind of `#define` for what these different return values mean?

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/WorkInProgress/MechanicMadness.dm (hunk 2 above)
Comment (Arivias) [reply code/WorkInProgress/MechanicMadness.dm:1540]: Good idea. Where would be the best place to put the defines?

Comment (ZeWaka) [reply code/WorkInProgress/MechanicMadness.dm:1540]: I guess a new file in std/defines?

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**